ext/winevt/winevt_query.c

```diff
@@ -120,15 +121,24 @@ rb_winevt_query_set_timeout(VALUE self, VALUE timeout)
 static VALUE
 rb_winevt_query_next(VALUE self)
 {
-  EVT_HANDLE event;
+  EVT_HANDLE hEvents[QUERY_ARRAY_SIZE];
   ULONG count;
+  DWORD status = ERROR_SUCCESS;
   struct WinevtQuery* winevtQuery;
 
   TypedData_Get_Struct(self, struct WinevtQuery, &rb_winevt_query_type, winevtQuery);
 
-  if (EvtNext(winevtQuery->query, 1, &event, INFINITE, 0, &count) != FALSE) {
-    winevtQuery->event = event;
+  if (!EvtNext(winevtQuery->query, QUERY_ARRAY_SIZE, hEvents, INFINITE, 0, &count) != FALSE) {
```

Comment:
Please, please remove the trailing `!= FALSE` :-1:

Reply:
Ouch! :confounded: